### PR TITLE
add annotation and prop tests

### PR DESCRIPTION
**Pull request by liangchenye**
test cases for AnnotationsKeyReversedDomain, AnnotationsKeyReservedNS,
AnnotationsKeyIgnoreUnknown, ExtensibilityIgnoreUnknownProp and
ValidValues

Signed-off-by: Liang Chenye <liangchenye@huawei.com>

### CODE REVIEW

**Comment by liangchenye** [comment]
- add a random key to test 'AnnotationsKeyIgnoreUnknown'
- add an 'Unknown' proper to `config.json` to test 'ExtensibilityIgnoreUnknownProp'
- using invalid version to test 'ValidValues'
   but seems runc does not popup errors.

**Comment by liangchenye** [comment]
rebased, PTAL @q384566678

File: validation/misc_props.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	}
+
+	for _, c := range cases {
+		config := util.LifecycleConfig{
+			BundleDir: bundleDir,
+			Actions:   c.action,
+			PreCreate: func(r *util.Runtime) error {
+				r.SetID(containerID)
+				saveConfig(configFile, c.eSpec)
+				return nil
```
Comment (q384566678) [inline validation/misc_props.go]: I think it's better to `return saveConfig(configFile, c.eSpec)`

**Comment by q384566678** [comment]
It was in a panic when I was testing. Here is the test. I'm looking for the reason.

```
➜  runtime-tools git:(4826ab8) ✗ ./oci-runtime-tool validate
key  SHOULD be named using a reverse domain notation
Refer to: https://github.com/opencontainers/runtime-spec/blob/v1.0.0/config.md#annotations
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x575f02]

goroutine 1 [running]:
github.com/opencontainers/runtime-tools/vendor/github.com/hashicorp/go-multierror.(*Error).Error(0x0, 0x9eaea0, 0x0)
	/home/zhouhao/workspace/Go/src/github.com/opencontainers/runtime-tools/vendor/github.com/hashicorp/go-multierror/multierror.go:15 +0x22
github.com/opencontainers/runtime-tools/vendor/github.com/urfave/cli.HandleExitCoder(0x9eaea0, 0x0)
	/home/zhouhao/workspace/Go/src/github.com/opencontainers/runtime-tools/vendor/github.com/urfave/cli/errors.go:102 +0xff
github.com/opencontainers/runtime-tools/vendor/github.com/urfave/cli.Command.Run(0x801238, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x807013, 0x16, 0x0, ...)
	/home/zhouhao/workspace/Go/src/github.com/opencontainers/runtime-tools/vendor/github.com/urfave/cli/command.go:196 +0xadc
github.com/opencontainers/runtime-tools/vendor/github.com/urfave/cli.(*App).Run(0xc420086b60, 0xc42000c080, 0x2, 0x2, 0x0, 0x0)
	/home/zhouhao/workspace/Go/src/github.com/opencontainers/runtime-tools/vendor/github.com/urfave/cli/app.go:250 +0x758
main.main()
	/home/zhouhao/workspace/Go/src/github.com/opencontainers/runtime-tools/cmd/oci-runtime-tool/main.go:50 +0x404
```

**Comment by liangchenye** [comment]
@q384566678 it works in my test, can you post your 'config.json' file?

**Comment by liangchenye** [comment]
Very strange, if I add the following `useless` lines, it works. I'll continue to debug it tomorrow.
```
			if levelErrors.Error == nil {
				return nil
			}
 			return levelErrors.Error
```

**Comment by liangchenye** [comment]
The original code return `*error`  instread of `error`. The new commit solves that.
It is more accurate to return `levelErrors.Error.ErrorOrNil()`. It works in my test.
PTAL @q384566678

File: validation/misc_props.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	bundleDir, err := util.PrepareBundle()
+	if err != nil {
+		util.Fatal(err)
+	}
+	defer os.RemoveAll(bundleDir)
+	configFile := filepath.Join(bundleDir, "config.json")
+
+	type extendedSpec struct {
+		rspecs.Spec
+		Unknown string `json:"unknown,omitempty"`
```
Comment (q384566678) [inline validation/misc_props.go]: I don't quite understand what that does.

Comment (liangchenye) [reply validation/misc_props.go]: I use this to test ’runtimes that are reading or processing this configuration file MUST NOT generate an error if they encounter an unknown property‘.  'Unknown' here (in my commit) means the 'unknown property', so when a runtime reads this, it does not know what it means and it should not generete an error.

**Comment by q384566678** [comment]
LGTM

**Comment by alban** [comment]
> using invalid version to test 'ValidValues' but seems runc does not popup errors.

I filed an issue about this: https://github.com/opencontainers/runc/issues/1846

**Pull request merged**